client/rest/src/test/java/org/elasticsearch/client/RestClientSingleHostTests.java

```diff
@@ -86,39 +90,49 @@
     private RestClient restClient;
     private Header[] defaultHeaders;
     private HttpHost httpHost;
-    private CloseableHttpClient httpClient;
-    private TrackingFailureListener failureListener;
+    private CloseableHttpAsyncClient httpClient;
+    private HostsTrackingFailureListener failureListener;
 
     @Before
+    @SuppressWarnings("unchecked")
     public void createRestClient() throws IOException {
-        httpClient = mock(CloseableHttpClient.class);
-        when(httpClient.execute(any(HttpHost.class), any(HttpRequest.class))).thenAnswer(new Answer<CloseableHttpResponse>() {
-            @Override
-            public CloseableHttpResponse answer(InvocationOnMock invocationOnMock) throws Throwable {
-                HttpUriRequest request = (HttpUriRequest) invocationOnMock.getArguments()[1];
-                //return the desired status code or exception depending on the path
-                if (request.getURI().getPath().equals("/soe")) {
-                    throw new SocketTimeoutException();
-                } else if (request.getURI().getPath().equals("/coe")) {
-                    throw new ConnectTimeoutException();
-                }
-                int statusCode = Integer.parseInt(request.getURI().getPath().substring(1));
-                StatusLine statusLine = new BasicStatusLine(new ProtocolVersion("http", 1, 1), statusCode, "");
+        httpClient = mock(CloseableHttpAsyncClient.class);
```

Comment:
yea, lots of it but not everything... I thought about it and it seemed that it would need to become more generic hence more complex. I will give it another try.